Core/clim-core/frames/frames.lisp

```diff
@@ -241,9 +232,20 @@ documentation produced by presentations.")
 (defmethod initialize-instance :after ((obj standard-application-frame)
                                        &key (icon nil icon-supplied-p)
                                        &allow-other-keys)
-  (setf (slot-value obj 'icon) (if icon-supplied-p
-                                   (%coerce-to-icon icon)
-                                   nil))
+  (labels ((coerce-to-icon (thing)
+             (typecase thing
+               ((or string pathname)
+                (make-pattern-from-bitmap-file thing))
+               (sequence
+                (map 'list #'coerce-to-icon thing))
+               (t
+                thing))))
+    (setf (slot-value obj 'icon) (cond ((not icon-supplied-p)
+                                        nil)
+                                       ((null icon)
+                                        nil)
+                                       (t
+                                        (coerce-to-icon icon)))))
```

Comment:
this seems bogus:

(cond ((not supplied-p) nil)
          ((null icon) nil)
          (t …))

is equivalent to

(and icon (coerce-to-icon icon))

or `(if (null icon) nil (coerce-to-icon icon))`

Also I'd rather have this change to initialize-instance squashed to the previous commit where coerce-to-icon was introduced (i.e I don't see a purpose in having that change split in two commits).

Reply:
I see that it is fixed in later commit, so I'm merging this PR.